File: Algorithms/p13_Roman_to_Integer_2.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        s_len = len(s)
        l = [0]*s_len
        for i in range(0,s_len):
            if s[i]=='I': l[i] = 1
            elif s[i]=='V': l[i] = 5
            elif s[i]=='X': l[i] = 10
            elif s[i]=='L': l[i] = 50
            elif s[i]=='C': l[i] = 100
            elif s[i]=='D': l[i] = 500
            elif s[i]=='M': l[i] = 1000
            else: l[i] = 0
        for i in range(1,s_len):
            if l[i]>l[i-1]:
                l[i-1] = -l[i-1]
        return sum(l)

File: Algorithms/p13_Roman_to_Integer_2.cpp
class Solution {
public:
    int romanToInt(string s) {
        int len = s.size();
        vector<int> l(len);
        for(int i=0;i<len;i++){
            switch(s[i]){
                case 'I': l[i] = 1; break;
                case 'V': l[i] = 5; break;
                case 'X': l[i] = 10; break;
                case 'L': l[i] = 50; break;
                case 'C': l[i] = 100; break;
                case 'D': l[i] = 500; break;
                case 'M': l[i] = 1000; break;
                default:  l[i] = 0;
            }
        }
        for(int i=1;i<len;i++){
            if(l[i]>l[i-1]){
                l[i-1] = -l[i-1];
            }
        }
        return accumulate(l.begin(),l.end(),0);
    }
};
